### ISSUE TITLE
why do we need to do a manual bump in pallet-salary?

### ISSUE THREAD

**Issue by xlc** [paritytech/polkadot-sdk#4545]
why we need someone to call bump to bump the cycle instead of schedule it in schduler?

**Comment by Doordashcon** [comment]
So add up the `RegistrationPeriod` and `PayoutPeriod` in the `on_initialize` hook and when this passes from the current cycle it will call `bump` on the next block? Doesn't seem like there are edge cases where doing this might fail?

**Comment by Doordashcon** [comment]
Would like to be assigned to this.

**Comment by Doordashcon** [comment]
We don't need an extrinsic to manually call `bump` then? this can be changed into an helper function but retain the `WeightInfo` for `bump`.

**Comment by ggwpez** [comment]
You can call the bump automatically in `on_idle` or `on_poll`. Yea weight still needs to be accounted for.

**Comment by bkchr** [comment]
We should use a `task` for this or not?

**Comment by gupnik** [comment]
> We should use a `task` for this or not?

Yes, that should work. Let me know if you face any issues @Doordashcon.

**Comment by bkchr** [comment]
@gupnik can you please post a link to an example on how to do this?

**Comment by gupnik** [comment]
> @gupnik can you please post a link to an example on how to do this?

Yes, [here](https://github.com/paritytech/polkadot-sdk/blob/ac98bc3fa158aa01faa3c0f95d10c8932affbcc2/substrate/frame/examples/tasks/src/lib.rs#L53) it is.

**Comment by gupnik** [comment]
> > @gupnik can you please post a link to an example on how to do this?
> 
> Yes, [here](https://github.com/paritytech/polkadot-sdk/blob/ac98bc3fa158aa01faa3c0f95d10c8932affbcc2/substrate/frame/examples/tasks/src/lib.rs#L52) it is, with the off-chain worker submitting the tasks [here](https://github.com/paritytech/polkadot-sdk/blob/ac98bc3fa158aa01faa3c0f95d10c8932affbcc2/substrate/frame/examples/tasks/src/lib.rs#L72).
